starport/pkg/cosmosclient/cosmosclient.go

```diff
@@ -384,7 +350,33 @@ func (c *Client) makeSureAccountHasTokens(ctx context.Context, address string) e
 		}
 	}
 
-	return nil
+	// make sure funds are retrieved.
+	ctx, cancel := context.WithTimeout(ctx, time.Minute)
+	defer cancel()
+
+	return backoff.Retry(func() error {
+		return c.checkAccountBalance(ctx, address)
+	}, backoff.WithContext(backoff.NewConstantBackOff(time.Second), ctx))
+}
+
+func (c *Client) checkAccountBalance(ctx context.Context, address string) (err error) {
+	balancesResp, err := banktypes.NewQueryClient(c.Context).AllBalances(ctx, &banktypes.QueryAllBalancesRequest{
+		Address: address,
+	})
+	if err != nil {
+		return err
+	}
+
+	// if the balance is enough do nothing.
+	if len(balancesResp.Balances) > 0 {
+		for _, coin := range balancesResp.Balances {
+			if coin.Denom == c.faucetDenom && coin.Amount.Uint64() >= c.faucetMinAmount {
+				return nil
+			}
+		}
+	}
+
+	return errors.New("account has not enough balance")
```

Comment:
```suggestion
	// if the balance is enough do nothing.
	if len(balancesResp.Balances) == 0 {
		return errors.New("account has not enough balance")
	}

	for _, coin := range balancesResp.Balances {
		if coin.Denom == c.faucetDenom &&
			coin.Amount.Uint64() >= c.faucetMinAmount {
			break
		}
	}
	return
```

Only a suggestion. I think the code can be cleaner and without too much indentation to the right, and it also avoids returning an error by default. I think it's easier to maintain.

Reply:
I agree bu, we still need to return with an error if there are balances but not enough of them. So in this case we need to return the error in two different places, kinda look harder to read to me.

Reply:
Ohh, I got it, this can solve the issue, but still more a suggestion, for me this is cleaner to read

```go
	// if the balance is enough do nothing.
	if len(balancesResp.Balances) == 0 {
		return errors.New("account has not enough balance")
	}
	for _, coin := range balancesResp.Balances {
		if coin.Denom == c.faucetDenom &&
			coin.Amount.Uint64() >= c.faucetMinAmount {
			return
		}
	}
	return errors.New("account has not enough balance")
```